File: fstBookstore/fst_bookstore_frontend/src/services/CartService.js
import {postRequest, postRequest_noReturn} from "../utils/ajax";

export const getCartItems = (user_id, callback) => {
    const url = '/getCartItems';
    const data = {userId: user_id};
    postRequest(url, data, callback);
}

export const updateItemNum = (userId, itemId, itemNum) => {
    const url = '/updateItemNum';
    const data = {
        itemNum: itemNum,
        userId: userId,
        itemId: itemId,
    };
    postRequest_noReturn(url, data);
}

export const deleteItem = (userId, itemId) => {
    const url = '/deleteItem';
    const data = {
        userId: userId,
        itemId: itemId,
    };
    postRequest_noReturn(url, data);
}

export const addItem = (userId, itemId) => {
    const url = '/addItem';
    const data = {
        userId: userId,
        itemId: itemId,
    };
    postRequest_noReturn(url, data);
}

File: bookstore_backend/src/main/java/com/bookstore/service/CartService.java
package com.bookstore.service;

import com.bookstore.entity.CartItem;
import java.util.List;

public interface CartService {
    List<CartItem> getCartItems(int userId);
    void updateItemNum(int itemNum, int userId, int itemId);
    void deleteItem(int userId, int itemId);
    void addItem(int userId, int itemId);
}
